backend/project-management-system/src/main/java/org/egov/works/validator/ProjectValidator.java

```diff
@@ -0,0 +1,301 @@
+package org.egov.works.validator;
+
+import com.jayway.jsonpath.JsonPath;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.tracer.model.CustomException;
+import org.egov.works.util.LocationUtil;
+import org.egov.works.util.MDMSUtils;
+import org.egov.works.web.models.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.egov.works.util.ProjectConstants.*;
+
+@Component
+@Slf4j
+public class ProjectValidator {
+
+    @Autowired
+    MDMSUtils mdmsUtils;
+
+    @Autowired
+    LocationUtil locationUtil;
+
+    public void validateCreateProjectRequest(ProjectRequest request) {
+        Map<String, String> errorMap = new HashMap<>();
+        List<Project> projects = request.getProjects();
+        RequestInfo requestInfo = request.getRequestInfo();
+
+        validateRequestInfo(requestInfo, errorMap);
+        validateProjectRequest(projects, errorMap);
+
+        String tenantId = projects.get(0).getTenantId();
+        String rootTenantId = tenantId.split("\\.")[0];
+
+        Object mdmsData = mdmsUtils.mDMSCall(request, rootTenantId);
+
+        validateMDMSData(projects, mdmsData,  errorMap);
+
+        List<String> locationsForValidation = getLocationForValidation(projects);
+        if (locationsForValidation.size() > 0)
+            validateLocation(locationsForValidation, tenantId, requestInfo, errorMap);
+
+        if (!errorMap.isEmpty())
+            throw new CustomException(errorMap);
+    }
+
+    public void valiateSearchProject(ProjectRequest project, Integer limit, Integer offset, String tenantId) {
```

Comment:
Please add comments for each method. What exactly does this method do? Even a one liner description should be good.

Reply:
Comments added for each method